### ISSUE TITLE
network.all_drop shows nothing when import a slim model

### ISSUE THREAD

**Issue by benwu232** [zsdonghao/tensorlayer#75]
When I import a slim model, e.g. vgg16, network.all_drop shows {}, but there should be some dropout layers in the model. Is this a bug?

Thanks a lot!

**Comment by zsdonghao** [comment]
@benwu232 It is correct.
tf-slim don't use placeholder to control the keep probs, it equivalent to 
```python
if is_train:
    network = DropoutLayer(network, 0.8, is_fix=True, name='xxx')
```
then by setting  `is_train` you control training and testing phase

**Comment by benwu232** [comment]
@zsdonghao  Thanks for the answer!

is_train can't be changed in a model when set. I think that's why you maintain a drop_dict to differentiate  training and testing.

Another answer, I guess, is saving the checkpoint and loading it while setting the is_train, which is a little bit tedious.

**Comment by zsdonghao** [comment]
@benwu232 sorry for the late reply...

Yes, it is the same idea. by defualt, tensorlayer uses placeholder to control the training and testing states ~  and all placeholders are stored in `network.all_drop`

**Comment by benwu232** [comment]
@zsdonghao  Thanks for your reply!

I agree with you. They are conceptually same.

Since in tensorflow, all noise layers, such as dropout, have the parameter is_training, I think it is easier to just pass the is_training to control all the noise layers.
